manager/dispatcher/assignments.go

```diff
@@ -245,3 +250,29 @@ func (a *assignmentSet) message() api.AssignmentsMessage {
 
 	return message
 }
+
+// secret populates the secret value from raft store. For external secrets, the value is populated
+// from the secret driver.
+func (a *assignmentSet) secret(readTx store.ReadTx, secretID string) (*api.Secret, error) {
+	secret := store.GetSecret(readTx, secretID)
+	if secret == nil {
```

Comment:
@cpuguy83 but if a driver was specified, I still need to query the `api.Secret` object to fetch the `api.Driver` (to correctly materialize the secret).